dataprofiler/tests/profilers/test_profiler_class_options.py

```diff
@@ -0,0 +1,110 @@
+from dataprofiler.profilers.profiler_options import ProfilerOptions
+from dataprofiler.tests.profilers.test_base_option import TestBaseOption
+
+
+class TestProfilerOptions(TestBaseOption):
+    
+    option_class = ProfilerOptions
+    keys = ["structured_options"]
+    
+    @classmethod
+    def get_options(self, **params):
+        options = ProfilerOptions()
+        options.set(params)
+        return options
+    
+    def test_init(self):
+        options = self.get_options()
+        for key in self.keys:
+            self.assertTrue(key in options.properties)
+
+    def test_set_helper(self):
+        super().test_set_helper()
+        option = self.get_options()
+        optpth = self.get_options_path()
+        
+        # Enable and Disable Option
+        for key in option.structured_options.properties:
+            option._set_helper({'structured_options.{}.is_enabled'.format(key): False}, '')
+            self.assertFalse(option.structured_options.properties[key].is_enabled)        
+            option._set_helper({'structured_options.{}.is_enabled'.format(key): True}, '')
+            self.assertTrue(option.structured_options.properties[key].is_enabled)        
+
+        # Treat is_enabled as a BooleanOption
+        for key in option.structured_options.properties:
+            expected_error = "type object 'structured_options.{}.is_enabled' has no attribute 'is_enabled'".format(key)
+            with self.assertRaisesRegex(AttributeError, expected_error):
+                option._set_helper({'structured_options.{}.is_enabled.is_enabled'.format(key): True}, '')    
```

Comment:
I might suggest having tests that cross multiple layers like this in its own test class as this seems to test more than just the profileroptions.

Also, not sure this needs to be in test_set_helper as `_set_helper` is called by `set` which is where the same test also exists and is more of a duplicate.

Reply:
Though I guess for each class we only make the helper. I'm open to ideas. Ultimately I don't think we need to duplicate the tests since the set is more like a wrapper which calls it.

However, i do think we need an overall api tests which does check multiple levels.

Reply:
That makes sense - it does test more than just the profiler options. The aim of this test was more to get the type object '...' has no attribute error from the BaseOption class. I didn't want to assume that the BooleanOption Test had been run so this would be the first time this error would be raised and checked for. 

And yes, the set and _set_helper are identical with the two differences being the function and variable path. I thought it might make sense to leave it this way to 
  - test that both return the same results when called the same way
  - explicitly write it out for anyone who may need to update this test in the future

Reply:
@JGSweets do you think that justification is sufficient or do you think I should delete most of the _set_helper tests?

Reply:
I think the PR is fine to merge, but I would suggest that since it is a unit test, we should be able to compartmentalize to the unit itself. BooleanOption test should be assumed to run, and a separate integration test should be able to catch if there's an issue when the two overlap.

Reply:
I think catching the attribute error could be validated in the location where the test for that validate occurs

Reply:
also might make the tests more modular based on keys etc